rust/arrow/src/array.rs

```diff
@@ -1034,6 +1034,35 @@ impl From<Vec<(Field, ArrayRef)>> for StructArray {
     }
 }
 
+impl From<(Vec<(Field, ArrayRef)>, Buffer, usize)> for StructArray {
+    fn from(triple: (Vec<(Field, ArrayRef)>, Buffer, usize)) -> Self {
+        let (field_types, field_values): (Vec<_>, Vec<_>) = triple.0.into_iter().unzip();
+
+        // Check the length of the child arrays
+        let length = field_values[0].len();
```

Comment:
Just a drive-by comment, but what happens if the vector is empty?

Reply:
I'd personally prefer panicking in this instance. I'd think it reasonable for the caller of the conversion method(s) to ensure they don't pass empty lists. We can at least `panic!("Field value list should not be empty when constructing struct array")`. We could use `TryFrom`, but I feel it would be overkill.

Reply:
That said, for the record, the C++ side allows 0-field struct arrays (even though they don't seem tested).

Reply:
Perhaps if an empty record batch and empty schema are fine (which I think both are), then we can/must allow there to be an empty struct array in general.
The conversion method in question might still need to panic, but I'll give this more thought.